rbc/typesystem.py

```diff
@@ -726,6 +726,14 @@ def __str__(self):
             return self.tostring()
         return tuple.__str__(self)
 
+    def __eq__(self, other: object) -> bool:
+        if not isinstance(other, self.__class__):
+            return False
+        return super().__eq__(other)
+
+    def __hash__(self) -> int:
+        return super().__hash__()
```

Comment:
Why do we need to define `__hash__` here while the base class (tuple) already provides it?

Reply:
Without it, the following fails:

```python
(Pdb++) hash(Type('float64'))
*** TypeError: unhashable type: 'Type'
```